components/automate-ui/src/app/entities/service-groups/service-groups.requests.ts

```diff
@@ -79,4 +90,34 @@ export class ServiceGroupsRequests {
 
     return this.httpClient.get<HealthSummary>(url);
   }
+
+  public getSuggestions(
+    field_name: string,
+    query_fragment: string,
+    filters: ServiceGroupFilters): Observable<any[]> {
+      const params = this.formatFilters(filters)
+        .set('field_name', field_name)
+        .set('query_fragment', query_fragment);
+      const url = `${APPLICATIONS_URL}/services-distinct-values`;
+
+      return this.httpClient.get<ServiceGroupSuggestion>(url, {params}).pipe(map(
+        (suggestions) => suggestions.values));
+  }
+
+  private formatFilters(filters: ServiceGroupFilters) {
+    let searchParam = new HttpParams();
+
+    if (filters.searchBar) {
+      searchParam = this.flattenSearchBar(filters.searchBar, searchParam);
+    }
+
+    return searchParam;
+  }
+
+  private flattenSearchBar(filters: object[], searchParam: HttpParams): HttpParams {
+    return reduce((params: HttpParams, filter: { type: string, text: string }) => {
```

Comment:
This filter type is used in about 8 places in the code base now. Would like to see you promote that to an exported interface so these couple lines could be a bit more clearly written:
```
  private flattenSearchBar(filters: FilterType[], searchParam: HttpParams): HttpParams {
    return reduce((params: HttpParams, filter: FilterType) => {
```